### PR TITLE
Event handler

### PR DESCRIPTION
**Pull request by itsUrcute**
added an event handler and did minor fixing around.
(PS: sorry for messy commit history, had to move os recently and then a lot of shit broke)

### CODE REVIEW

**Comment by TogarashiPepper** [comment]
Lgtm!

File: src/internals/commandHandler.ts [hunk 1]
```diff
@@ -84,7 +74,7 @@ function startHandler(client: Client): void {
     "ready",
     async() => {
       // Loading commands from /commands.
-      await loadCommands(commands);
+      await loadFiles<Command>(commands, "./commands");
       const s = commands.size === 1 ? "" : "s";
       console.log(`${commands.size} commands${s} loaded.`);
```
Comment (cheeze2000) [inline src/internals/commandHandler.ts:79]: you might get "commandss"

File: src/internals/commandHandler.ts [hunk 2]
```diff
@@ -84,7 +74,7 @@ function startHandler(client: Client): void {
     "ready",
     async() => {
       // Loading commands from /commands.
-      await loadCommands(commands);
+      await loadFiles<Command>(commands, "./commands");
```
Comment (cheeze2000) [inline src/internals/commandHandler.ts:77]: please don't use relative paths

File: src/utils/loadFiles.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { readdir } from "fs/promises";
+import type { Collection } from "discord.js";
+import type { DiscordEvent } from "../types/DiscordEvent";
+import type { Command } from "../types/Command";
+
+async function loadFiles<T extends Command | DiscordEvent>(collection: Collection<string, T>, path = "./commands"): Promise<void> {
+  const files = await readdir(path, { withFileTypes: true });
+  for (const element of files) {
+    if (element.isFile() && element.name.endsWith(".js")) {
+      const imp: {default: T} = await import(`.${path}/${element.name}`);
```
Comment (cheeze2000) [inline src/utils/loadFiles.ts:10]: use the path api. do something like 
`import { join } from "path";`
and then
`join(path, element.name);`

File: src/utils/loadFiles.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      collection.set(
+        imp.default.name.toLowerCase(),
+        imp.default
+      );
+    }
+    else if (element.isDirectory()) {
+      await loadFiles(collection, `${path}/${element.name}`);
```
Comment (cheeze2000) [inline src/utils/loadFiles.ts:17]: same comment as above

File: src/utils/loadFiles.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { readdir } from "fs/promises";
+import type { Collection } from "discord.js";
+import type { DiscordEvent } from "../types/DiscordEvent";
+import type { Command } from "../types/Command";
+
+async function loadFiles<T extends Command | DiscordEvent>(collection: Collection<string, T>, path = "./commands"): Promise<void> {
```
Comment (cheeze2000) [inline src/utils/loadFiles.ts:6]: is there really a need for default argument?

File: src/internals/eventHandler.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import { Collection } from "discord.js";
+import { loadFiles } from "../utils/loadFiles";
+import type { Client, ClientEvents } from "discord.js";
+import type { DiscordEvent } from "../types/DiscordEvent";
+
+const events = new Collection<string, DiscordEvent>();
+
+async function startEventHandler(client: Client): Promise<void> {
+  await loadFiles<DiscordEvent>(events, "./events");
```
Comment (cheeze2000) [inline src/internals/eventHandler.ts:9]: refactor

**Review by cheeze2000 — CHANGES_REQUESTED** [review]

File: src/events/general/ready.ts [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { DiscordEvent } from "../../types/DiscordEvent";
+
+const event: DiscordEvent = {
+  name: "a",
+  event: "ready",
+  description: "boop",
+  once: true,
+  async execute() {
+    // big pogger
```
[the remark this answers is not in the document]
Comment (itsUrcute) [reply src/events/general/ready.ts:9]: big pogger

File: src/internals/commandHandler.ts (hunk 1 above)
Comment (itsUrcute) [reply src/internals/commandHandler.ts:79]: extra s oops. will fix

File: src/utils/loadFiles.ts (hunk 5 above)
Comment (itsUrcute) [reply src/utils/loadFiles.ts:6]: nope, will be removed.

File: src/commands/general/ping.ts [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import type { Command } from "../../types/Command";
```
Comment (cheeze2000) [inline src/commands/general/ping.ts:1]: you have custom import paths like `#types` in your package.json, use it

File: src/internals/eventHandler.ts [hunk 9]
```diff
@@ -0,0 +1,38 @@
+import { Collection } from "discord.js";
+import { loadFiles } from "../utils/loadFiles";
+import { join } from "path";
+import type { Client, ClientEvents } from "discord.js";
+import type { DiscordEvent } from "../types/DiscordEvent";
+
+const events = new Collection<string, DiscordEvent>();
```
Comment (cheeze2000) [inline src/internals/eventHandler.ts:7]: what is the point of putting `events` outside? why can't events be declared via line 10?

File: src/utils/loadFiles.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+import { join } from "path";
+import type { Collection } from "discord.js";
+import type { DiscordEvent } from "../types/DiscordEvent";
+import type { Command } from "../types/Command";
+
+async function loadFiles<T extends Command | DiscordEvent>(collection: Collection<string, T>, path: string): Promise<void> {
+  const files = await readdir(path, { withFileTypes: true });
+  for (const element of files) {
+    if (element.isFile() && element.name.endsWith(".js")) {
+      const imp: {default: T} = await import(join(path, element.name));
```
Comment (cheeze2000) [inline src/utils/loadFiles.ts:11]: `{ default: T }` not sure why eslint doesn't care about this

**Review by cheeze2000 — CHANGES_REQUESTED** [review]

File: src/internals/eventHandler.ts (hunk 9 above)
Comment (itsUrcute) [reply src/internals/eventHandler.ts:7]: the collection was defined outside with future needs for exporting the event in mind

File: src/utils/loadFiles.ts [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+      collection.set(
+        imp.default.name.toLowerCase(),
+        imp.default
+      );
+    }
+    else if (element.isDirectory()) {
```
Comment (cheeze2000) [inline src/utils/loadFiles.ts:17]: brace style is inconsistent with [commandHandler.ts](https://github.com/itsUrcute/algo-chan/blob/eventHandler/src/internals/commandHandler.ts). please look through

File: src/commands/general/ping.ts (hunk 8 above)
Comment (itsUrcute) [reply src/commands/general/ping.ts:1]: future PR

**Pull request merged**